### PR TITLE
Update Code to 1.81.1

### PR DESCRIPTION
**Pull request by benz0li**
Ping @code-asher

### CODE REVIEW

**Comment by benz0li** [comment]
Increase of build timeout is not required. It took something over 30 mins on my machine.

ℹ️ The pipeline for https://github.com/coder/code-server/pull/6369 failed due to an e2e test error.

File: patches/base-path.diff [hunk 1]
```diff
@@ -255,15 +255,7 @@ Index: code-server/lib/vscode/src/vs/code/browser/workbench/workbench.ts
 ===================================================================
 --- code-server.orig/lib/vscode/src/vs/code/browser/workbench/workbench.ts
 +++ code-server/lib/vscode/src/vs/code/browser/workbench/workbench.ts
-@@ -484,6 +484,7 @@ function doCreateUri(path: string, query
- 		});
- 	}
- 
-+	path = (window.location.pathname + "/" + path).replace(/\/\/+/g, "/")
```
Comment (code-asher) [inline patches/base-path.diff:262]: I am not sure the authentication callbacks are still used in code-server, but we might want to add this back.  Looks like it is found on line 147 now.

**Comment by code-asher** [comment]
Weird, I am not sure why the e2e tests are failing.  I re-ran them a few times but it seems like code-server is not loading in time, and I am seeing similar issues locally, but the really weird thing is that it only happens if I have the browser dev tools open.  Maybe they added some kind of debug tool and it runs with dev tools open or in the e2e tests for some reason.

I will be able to investigate more next week.

File: patches/base-path.diff (hunk 1 above)
Comment (benz0li) [reply patches/base-path.diff:262]: The following commits had a significant impact on patches:

* https://github.com/microsoft/vscode/commit/322bc2d7d86e8d0c2ccb1bfdabe594e2011d500f
  * File: `lib/vscode/src/vs/code/browser/workbench/workbench.ts`
  * Patches: `base-path.diff`, `proxy-uri.diff`
* https://github.com/microsoft/vscode/commit/ef42b49e806b4a4f1202ec2e31f441465a7047de
  * File: `lib/vscode/src/vs/server/node/serverServices.ts`
  * Patch: `telemetry.diff`

I am not sure I updated the patches correctly.

**Comment by benz0li** [comment]
> I will be able to investigate more next week.

Thank you. It would be great to have a release by the end of this week.

**Comment by benz0li** [comment]
Ping @code-asher

**Comment by code-asher** [comment]
I have been trying to reproduce the hang but I cannot seem to get it to happen again.

I updated the callback path to account for the base path.

In addition to the above, I also tested display languages and the github token.

I get a lot (~23) of "throttler is disposed" errors when closing the browser but it does not seem to cause any issues so I think we can ignore it.  There is an open issue for it: https://github.com/microsoft/vscode/issues/188929

Assuming CI passes I think we will be good to go.

**Comment by code-asher** [comment]
e2e tests still timing out unfortunately.

**Comment by code-asher** [comment]
Sorted the test hang, but now there seems to be some syntax issue preventing WebKit browsers from working.

**Comment by code-asher** [comment]
Thank you for experimenting with it.

I ran the CI build in Safari and the syntax error seems to be coming from some `static { }` blocks.  I guess Safari does not like these.

They are coming from here: https://github.com/microsoft/vscode/blob/6c3e3dba23e8fadc360aed75ce363ba185c49794/extensions/notebook-renderers/src/color.ts#L895-L902

It does not seem like this file has changed recently, so I wonder if the compilation target was changed and now it is outputting JavaScript that is not quite compatible with Safari.

**Comment by code-asher** [comment]
Oh, to clarify the submodule, the e2e step downloads the build package from CI and uses that for the tests, so there is no need to populate lib/vscode.

**Comment by code-asher** [comment]
In the 1.81.1 `static readonly white = new Color(new RGBA(255, 255, 255, 1))` gets transpiled into: 
```js
static {
  this.white = new I(new E(255, 255, 255, 1))
}
```

In the previous version we got:
```js
I.white = new I(new E(255, 255, 255, 1))
```

**Comment by benz0li** [comment]
> I get the same error on vscode.dev so it appears to be an upstream issue.

No worries. We can also skip this version. If you feel the same, please close this issue.

I only prefer versions that work absolutely fine for all browsers.

**Comment by benz0li** [comment]
Let us focus on the next release, then: https://github.com/microsoft/vscode/issues/189778

**Comment by code-asher** [comment]
Sorry all, I was out last week so I was not able to get back to this.  I think skipping makes a lot of sense, we will need to see if we can get it patched upstream in the meantime if it has not already been.

**Pull request closed without merging**